src/olympia/activity/utils.py

```diff
@@ -63,55 +67,74 @@ def _extra_email_reply_parse(self, email):
             return split_email[0]
 
     def get_uuid(self):
-        for to in self.email.get('To', []):
+        to_header = self.email.get('To', [])
+        for to in to_header:
             address = to.get('EmailAddress', '')
             if address.startswith(self.address_prefix):
                 # Strip everything between "reviewreply+" and the "@" sign.
                 return address[len(self.address_prefix):].split('@')[0]
         log.exception(
             'TO: address missing or not related to activity emails. (%s)'
-            % self.email)
+            % to_header)
         raise ActivityEmailUUIDError(
-            'TO: address doesn\'t contain activity email uuid (%s)'
-            % self.email)
+            'TO: address doesn\'t contain activity email uuid (%s).'
+            % to_header)
 
-    def get_body(self):
-        return self.reply
 
+def add_email_to_activity_log_wrapper(message):
+    note = None
+    reason = 'Undefined Error.'
+    try:
+        parser = ActivityEmailParser(message)
+        note = add_email_to_activity_log(parser)
+    except ActivityEmailError as exception:
+        reason = str(exception)
+
+    if not note:
+        try:
+            bounce_mail(message, reason)
+        except:
+            log.error('Bouncing invalid email failed.')
+    return note
 
-def add_email_to_activity_log(message):
-    log.debug("Saving from email reply")
 
+def add_email_to_activity_log(parser):
+    log.debug("Saving from email reply")
+    uuid = parser.get_uuid()
     try:
-        parser = ActivityEmailParser(message)
-        uuid = parser.get_uuid()
         token = ActivityLogToken.objects.get(uuid=uuid)
     except ActivityLogToken.DoesNotExist:
         log.error('An email was skipped with non-existing uuid %s.' % uuid)
-        return False
-    except ActivityEmailError:
-        # We logged already when the exception occurred.
-        return False
+        raise ActivityEmailUUIDError(
+            'UUID found in email address TO: header but is not a valid token '
+            '(%s).' % uuid)
 
     version = token.version
     user = token.user
     if token.is_valid():
         log_type = action_from_user(user, version)
 
         if log_type:
-            note = log_and_notify(log_type, parser.get_body(), user, version)
+            note = log_and_notify(log_type, parser.reply, user, version)
             log.info('A new note has been created (from %s using tokenid %s).'
                      % (user.id, uuid))
             token.increment_use()
             return note
         else:
             log.error('%s did not have perms to reply to email thread %s.'
                       % (user.email, version.id))
+            raise ActivityEmailTokenError(
+                'You don\'t have permission to reply to this add-on.  You '
```

Comment:
http://www.theatlantic.com/entertainment/archive/2011/01/you-can-have-my-double-space-when-you-pry-it-from-my-cold-dead-hands/69592/